.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,135 @@
+                echo "${{ secrets.UPLOAD_BENCHMARK_RESULTS }}" > ~/.ssh/github_actions
+                chmod 600 ~/.ssh/github_actions
+                ssh-agent -a $SSH_AUTH_SOCK > /dev/null
+                ssh-add ~/.ssh/github_actions
+        -   name: Pull previous results from benchmarks repo
+            env:
+                SSH_AUTH_SOCK: /tmp/ssh_agent_benchmarks.sock
+            run: |
+                git clone git@github.com:nilearn/benchmarks.git benchmarks_repo
+                cp -r benchmarks_repo/results .
+        -   name: Get all the machine info
+            run: |
+                asv machine --yes
+        -   name: Edit asv-machine.json to a custom machine name
+            run: python ./build_tools/github/set_machine_name.py fv-az1113-357
+        -   name: Run all benchmarks on the latest commit
+            run: |
+                asv run -ev --append-samples --machine fv-az1113-357
+        -   name: Create html with all results
+            run: |
+                asv publish
+        -   uses: actions/upload-artifact@v4
+            with:
+                name: Upload asv benchmark results as artifacts
+                path: |
+                    ./env
+                    ./html
+                    ./results
+                compression-level: 9
+        -   name: Push new results back to benchmarks repo
+            env:
+                SSH_AUTH_SOCK: /tmp/ssh_agent.sock
+            run: |
+                cd benchmarks_repo
+                cp -r ../results .
+                cp -r ../html/* .
+                git config --global user.name "GitHub Actions"
+                git config --global user.email "actions@github.com"
+                git add .
+                git commit -m "Update benchmark results and HTML"
+                git push origin main
+                cd ..
+        -   name: Add SSH key for nilearn.github.io repo
+            env:
+                SSH_AUTH_SOCK: /tmp/ssh_agent_nilearn_github_io.sock
+            run: |
+                mkdir -p ~/.ssh
+                ssh-keyscan github.com >> ~/.ssh/known_hosts
+                echo "${{ secrets.ACTIONS_SSH_DEPLOY }}" > ~/.ssh/github_actions
+                chmod 600 ~/.ssh/github_actions
+                ssh-agent -a $SSH_AUTH_SOCK > /dev/null
+                ssh-add ~/.ssh/github_actions
+        -   name: Deploy updated benchmarks to nilearn.github.io/benchmarks
+            env:
+                SSH_AUTH_SOCK: /tmp/ssh_agent_nilearn_github_io.sock
+            run: |
+                git clone --depth 1 git@github.com:nilearn/nilearn.github.io.git --recursive
+                cd nilearn.github.io
+                git config --global user.name "GitHub Actions"
+                git config --global user.email "actions@github.com"
+                git submodule update --remote --merge
+                git add benchmarks
+                git commit -m "Deploy benchmarks"
+                git push origin main
```

Comment:
quick suggestion:
- let's not submodule things in the doc repo
- let's deploy directly to github pages from the benchmark repo
- in both cases the URL will be: https://nilearn.github.io/benchmarks/
- this way we avoid having a submodule and keep it updated (will also make the history of the doc repo shorter: https://github.com/nilearn/nilearn.github.io/commits/main/)

Reply:
> in both cases the URL will be: https://nilearn.github.io/benchmarks/

I see. I did not know this. Let's try that!